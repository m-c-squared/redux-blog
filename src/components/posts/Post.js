import React from 'react';
import PropTypes from 'prop-types';

const Post = ({ post }) => {  
  return (
    <>
    <li>
      <h2>{post.userId}</h2>
      <h3>{post.title}</h3>
      <h4>{post.body}</h4>
    </li>
    </>
  );
};

Post.propTypes =  {
  post: PropTypes.object.isRequired
};
export default Post;
