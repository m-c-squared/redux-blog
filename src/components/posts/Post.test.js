import React from 'react';
import renderer from 'react-test-renderer';
import Post from './Post';
import { MemoryRouter } from 'react-router-dom';

describe('Post', () => {
  it('matches a snapshot', () => {
    const post = {
      userId: '5', 
      title: 'sunt aut facere repellat provident occaecati excepturi optio reprehenderit',
      body: 'quia et suscipit suscipit recusandae consequuntur expedita et cum reprehenderit molestiae ut ut quas totam nostrum rerum est autem sunt rem eveniet architecto'
    };
    const tree = renderer.create(
      <MemoryRouter>
        <Post post={post}/>
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
