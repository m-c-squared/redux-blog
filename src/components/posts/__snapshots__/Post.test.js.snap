// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post matches a snapshot 1`] = `
<li>
  <h2>
    5
  </h2>
  <h3>
    sunt aut facere repellat provident occaecati excepturi optio reprehenderit
  </h3>
  <h4>
    quia et suscipit suscipit recusandae consequuntur expedita et cum reprehenderit molestiae ut ut quas totam nostrum rerum est autem sunt rem eveniet architecto
  </h4>
</li>
`;
